File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/ViewVoterAdapter.java
package com.scatl.uestcbbs.module.post.adapter;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.ViewVoterBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;

public class ViewVoterAdapter extends BaseQuickAdapter<ViewVoterBean, BaseViewHolder> {
    public ViewVoterAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, ViewVoterBean item) {
        helper.setText(R.id.name, item.name)
                .addOnClickListener(R.id.avatar);

        GlideLoader4Common.simpleLoad(mContext, item.avatar, helper.getView(R.id.avatar));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/ViewVoterAdapter.kt
package com.scatl.uestcbbs.module.post.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemViewVoterBinding
import com.scatl.uestcbbs.entity.PostDetailBean
import com.scatl.uestcbbs.entity.ViewVoterBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.Constant
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 14:04
 */
class ViewVoterAdapter: PreloadAdapter<ViewVoterBean, ItemViewVoterBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemViewVoterBinding {
        return ItemViewVoterBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemViewVoterBinding>, position: Int, item: ViewVoterBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }
        holder.binding.name.text = item.name
        holder.binding.avatar.load(Constant.USER_AVATAR_URL + item.uid)
    }
}